lib/js/caml_obj.js

```diff
@@ -119,35 +124,97 @@ function caml_compare(_a, _b) {
             var len_a = a.length | 0;
             var len_b = b.length | 0;
             if (len_a === len_b) {
-              var a$1 = a;
-              var b$1 = b;
-              var _i = 0;
-              var same_length = len_a;
-              while(true) {
-                var i = _i;
-                if (i === same_length) {
-                  return 0;
-                } else {
-                  var res = caml_compare(a$1[i], b$1[i]);
-                  if (res !== 0) {
-                    return res;
+              if (a.constructor === Object && b.constructor === Object) {
+                var a$1 = a;
+                var b$1 = b;
+                var min_key_lhs = [/* None */0];
+                var min_key_rhs = [/* None */0];
+                var do_key = function (param, key) {
+                  var min_key = param[2];
+                  var b = param[1];
+                  if (!b.hasOwnProperty(key) || caml_compare(param[0][key], b[key]) > 0) {
+                    var match = min_key[0];
+                    if (match) {
+                      if (key < match[0]) {
+                        min_key[0] = /* Some */[key];
+                        return /* () */0;
+                      } else {
+                        return 0;
+                      }
+                    } else {
+                      min_key[0] = /* Some */[key];
+                      return /* () */0;
+                    }
                   } else {
-                    _i = i + 1 | 0;
-                    continue ;
+                    return 0;
                   }
+                };
+                var partial_arg = /* tuple */[
+                  a$1,
+                  b$1,
+                  min_key_rhs
+                ];
+                var do_key_a = (function(partial_arg){
```

Comment:
Wondering whether it's possible to avoid creating a function here -- perhaps by writing the entire aux_obj_compare in js directly.